eth/client.go

```diff
@@ -875,12 +875,12 @@ func (c *client) CheckTx(tx *types.Transaction) error {
 		case err := <-txSub.Err():
 			return err
 		case receipt := <-receipts:
-			if tx.Hash() == receipt.TxHash {
+			if tx.Hash() == receipt.txOriginHash {
 				if receipt.err != nil {
 					return receipt.err
 				}
 				if receipt.Status == uint64(0) {
-					return fmt.Errorf("transaction failed txHash=%v", receipt.TxHash.Hex())
+					return fmt.Errorf("transaction failed txHash=%v", receipt.txOriginHash.Hex())
```

Comment:
I think we should log the tx hash attached to the receipt here because if the receipt indicates a failure then that means the tx associated with the receipt failed and not necessarily the original tx (since the receipt returned can be for a replacement tx and not for the original tx).